### PR TITLE
PEAR-1948: Update `TypeScript` to `5.6.3`

### PR DESCRIPTION
**Pull request by paribartandhakal**
## Description
1. Updated `TypeScript` from `4.6.4` to `5.6.3`.
2. Added required types.

## Checklist
- [ ] Added proper unit tests
- [ ] Left proper TODO messages for any remaining tasks
- [ ] Scanned for web accessibility with **aXe**, and mitigated or documented
      flagged issues

### CODE REVIEW

**Review by ocshawn — COMMENTED** [review]
On a clean install `npm run build` fails 
`npm run build:clean` works and then afterword `npm run build` works

`npm run dev` has lots of warnings may need to update @types devDependencies

File: packages/core/src/features/cohort/availableCohortsSlice.ts [hunk 1]
```diff
@@ -1033,3 +1031,65 @@ export const divideCurrentCohortFilterSetFilterByPrefix = (
 
   return divideFilterSetByPrefix(cohort?.filters, prefixes);
 };
+
```
Comment (amlehman) [inline packages/core/src/features/cohort/availableCohortsSlice.ts:1034]: Was this to fix the circular import warning?

File: packages/portal-proto/src/features/facets/DateRangeFacet.tsx [hunk 2]
```diff
@@ -71,7 +71,7 @@ const DateRangeFacet: React.FC<DateRangeFacetProps> = ({
       fromOp: ">=",
       toOp: "<=",
     };
-    console.log({ data });
+    // console.log({ data }); //TODO: confirm this should be removed
```
Comment (amlehman) [inline packages/portal-proto/src/features/facets/DateRangeFacet.tsx]: I would just remove this, no reason to leave this as a TODO.

**Review by amlehman — APPROVED** [review]

File: packages/portal-proto/src/features/facets/DateRangeFacet.tsx (hunk 2 above)
Comment (craigrbarnes) [reply packages/portal-proto/src/features/facets/DateRangeFacet.tsx]: I added the TODO and removed the commented-out console.log

**Review by craigrbarnes — APPROVED** [review]
Changes are fine

File: packages/core/src/features/cohort/availableCohortsSlice.ts (hunk 1 above)
Comment (paribartandhakal) [reply packages/core/src/features/cohort/availableCohortsSlice.ts:1034]: Yes. should've kept the const in different file.

Comment (paribartandhakal) [reply packages/core/src/features/cohort/availableCohortsSlice.ts:1034]: That warning was driving me crazy 😄

**Pull request merged**